pkg/controller/periodic/job.go

```diff
@@ -123,19 +123,20 @@ func (j *JobController) WaitForJobCompletion(ctx context.Context, job *batchv1.J
 			if !ok {
 				return fmt.Errorf("watcher channel was closed unexpectedly")
 			}
-			if event.Type == watch.Error {
-				return fmt.Errorf("watcher received error event: %v", event.Object)
+
+			if event.Type != watch.Modified {
+				continue
 			}
 
 			job := event.Object.(*batchv1.Job)
 			if job == nil {
-				return fmt.Errorf("cannot cast event objeto into job: %v", event.Object)
+				return fmt.Errorf("failed to cast job event: %v", event.Object)
 			}
 			if job.Status.Succeeded > 0 {
 				return nil
 			}
 			if job.Status.Failed > 0 {
-				return fmt.Errorf("job %s failed", job.Name)
+				return fmt.Errorf("job %s failed: %s", job.Name, job.Status.Conditions[0].Message)
```

Comment:
Sorry, but this doesn't make much sense to me. Why the first condition message? What if there are no condtions?